VistaCoreLibs/VistaKernel/OpenSG/VistaOpenSGDisplayBridge.cpp

```diff
@@ -512,3882 +494,3590 @@ std::string VistaOpenSGDisplayBridge::DisplayData::GetDisplayName() const
+  osg::ViewportPtr pOldPort = bRight ? m_RightViewport : m_Viewport;
+
+  if (bCopyData) {
+    beginEditCP(pNewPort);
+    {
+      pNewPort->setLeft(pOldPort->getLeft());
+      pNewPort->setBottom(pOldPort->getBottom());
+      pNewPort->setRight(pOldPort->getRight());
+      pNewPort->setTop(pOldPort->getTop());
+      pNewPort->setBackground(pOldPort->getBackground());
+      pNewPort->setCamera(pOldPort->getCamera());
+      pNewPort->setRoot(pOldPort->getRoot());
+      for (osg::UInt32 i = 0; i < pOldPort->getForegrounds().size(); ++i) {
+        pNewPort->getForegrounds().push_back(pOldPort->getForegrounds()[i]);
+      }
+    }
+    endEditCP(pNewPort);
+  }
+
+  osg::WindowPtr pParent = pOldPort->getParent();
+  beginEditCP(pParent, osg::Window::PortFieldMask);
+  pParent->replacePortBy(pOldPort, pNewPort);
+  endEditCP(pParent, osg::Window::PortFieldMask);
+
+  if (bRight) {
+    m_RightViewport = pNewPort;
+  } else {
+    m_Viewport = pNewPort;
+  }
+}
+
+void VistaOpenSGDisplayBridge::ViewportData::OnWindowResize(
+    VistaViewport* pViewport, const int nNewWindowWidth, const int nNewWindowHeight) {
+  if (m_bIsPerformingWindowResizeUpdate)
+    return;
+
+  m_bIsPerformingWindowResizeUpdate = true;
+  if (m_bResizeWithWindow) {
+    int nViewportWidth  = m_fRelativeViewportToWindowWidth * nNewWindowWidth;
+    int nViewportHeight = m_fRelativeViewportToWindowHeight * nNewWindowHeight;
+    pViewport->GetViewportProperties()->SetSize(nViewportWidth, nViewportHeight);
+  }
+  if (m_bRepositionWithWindow) {
+    int nViewportPosX = m_fRelativeViewportToWindowPosX * nNewWindowWidth;
+    int nViewportPosY = m_fRelativeViewportToWindowPosY * nNewWindowHeight;
+    pViewport->GetViewportProperties()->SetPosition(nViewportPosX, nViewportPosY);
+  }
+  m_bIsPerformingWindowResizeUpdate = false;
+}
+
+void VistaOpenSGDisplayBridge::ViewportData::SetupRenderTargets(
+    const int nWidth, const int nHeight) {
+  ClearRenderTargets(); // @IMGTODO: better: incremental update
+
+  CHECKFORGLERROR("SetupRenderTargets::OnEntry");
+
+  glewInit(); // @IMGTODO: where to put?
+  assert(glewGetExtension("GL_EXT_framebuffer_object") == GL_TRUE);
+  assert(__glewGenFramebuffersEXT != NULL);
+
+  CHECKFORGLERROR("SetupRenderTargets::PostGLEWInit");
+
+  int nCount = (m_bStereo ? 2 : 1);
+
+  glGenFramebuffers(nCount, m_nFBOId);
+  glGenRenderbuffers(nCount, m_nDepthBufferId);
+
+  CHECKFORGLERROR("SetupRenderTargets::PostGenBuffers");
+
+  for (int i = 0; i < nCount; ++i) {
+    if (m_bUseTextureSwapping) {
+      m_pReadTextureTarget[i] = new VistaGLTexture();
+      m_pReadTextureTarget[i]->Set2DData(nWidth, nHeight, NULL, false);
+      m_pReadTextureTarget[i]->SetMinFilter(GL_LINEAR);
+      m_pReadTextureTarget[i]->SetMagFilter(GL_LINEAR);
+    }
+    m_pWriteTextureTarget[i] = new VistaGLTexture();
+    m_pWriteTextureTarget[i]->Set2DData(nWidth, nHeight, NULL, false, GL_RGBA8, GL_RGBA, GL_UNSIGNED_BYTE);
```

Comment:
OpenVR accepts only GL Textures which are transferable with DXGI. These settings ensure that.